### ISSUE TITLE
Support libraries

### ISSUE THREAD

**Issue by jbojcic1** [manfredsteyer/ngx-build-plus#68]
Currently it seems like you can't add ngx-build-plus to the library generated with cli because it doesn't have serve command in the architect in angular.json

I tried setting it there manually but it doesn't seem to be working so I guess libraries are not supported at the moment?

**Comment by manfredsteyer** [comment]
Yes, that's right. The reason is that there is a compleatly different mechanism used for libs.

**Comment by manfredsteyer** [comment]
I'm curious: What's your use case?

**Comment by jbojcic1** [comment]
Currently Angular CLI doesn't support having global styles in the library so I wanted to extend webpack to bundle styles and copy them to dist folder on each build.

**Comment by jbojcic1** [comment]
As you can see here https://github.com/angular/angular-cli/issues/10869, lot of people have this issue. One of the cli team members has confirmed that they don't plan to support it and has recommended using this lib to customize.

**Comment by manfredsteyer** [comment]
I see. Could a post-build script do the job?

**Comment by chlab** [comment]
I also tried using ngx-build-plus with a library. One use case I'm interested is the styles building like @jbojcic1 and the other one would be to remove unused moment.js locales from my library build.

Do you see library support as something you might consider adding?

**Comment by jbojcic1** [comment]
> I see. Could a post-build script do the job?

Post build doesn't work with ng build watch mode. I was able to solve my issue using this library https://github.com/shlomiassaf/ng-cli-packagr-tasks

**Comment by qaxqaxqax** [comment]
I am also interested! I want to create a library where I can use worker-plugin to extend my build. The aim is to use "enjoyable" web workers from libraries.

**Comment by tommck** [comment]
Darn.. I was really excited to see this Issue, but now I'm pretty disappointed that it isn't supported (yet)

**Comment by loedeman** [comment]
Hmmm. I was also hoping ngx-build-plus could lend me a hand with libraries. The problem I am facing is that pre build transformations for TypeScript do not work when using ng-packagr. We do not use them an awful lot, but having nameof<X>() support is very important for us and until now not possible when compiling an Angular library. Do you plan to support libraries in the future?
